### ISSUE TITLE
Inline configuration errors not reported

### ISSUE THREAD

**Issue by jinglesthula** [AtomLinter/linter-eslint#1132]
## Issue Type
Bug

## Issue Description
If you put this comment at the top of a file, the linter encounters an error.
`/* eslint semi: ["error", "never", { "beforeStatementContinuationChars": "always"}] */`
The error is:
Severity: Error
Provider: ESLint
Description: Error while running ESLint: \<file path you add the comment to>\Foo.js line 1:.

Nothing is logged in the dev tools console, and the error reported in the linter UI appears to be an empty string.  Not sure how to diagnose further.  The `beforeStatementContinuationChars` feature was added recently in eslint, so I'm wondering if better error reporting when the package encounters an unknown bit in the eslint config/comments would help with diagnosing and reporting other such cases. 

Note the "object option when never" difference between these two docs versions:
[https://eslint.org/docs/4.0.0/rules/semi](https://eslint.org/docs/4.0.0/rules/semi)
[https://eslint.org/docs/rules/semi](https://eslint.org/docs/rules/semi)

As a side note, I also have prettier configured, and when it runs it reports this error:
```
C:\path\to\Foo.js line 1: Configuration for rule "semi" is invalid: Value "never,[object Object]" should NOT have more than 1 items. Value "never" should be equal to one of the allowed values. Value "never,[object Object]" should match some schema in anyOf.
```
My project itself has eslint 4.9.0 as a dependency, but I suspect the atom package is not looking at that.

## Bug Checklist
- [x] Restart Atom
- [x] Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- [x] Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Atom version: 1.26.0
linter-eslint version: 8.4.1
ESLint version: 4.9.0
Hours since last Atom restart: 0
Platform: win32
Using local project ESLint from: C:\<redacted path to project>\node_modules\eslint
Current file's scopes: [
  "source.js",
  "comment.block.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by jinglesthula** [comment]
Ah, gotcha.  I guess it does make more sense for the linter to use for a given project whatever the project itself uses for consistency between the editor and the actual eslint version that would be used in hooks or CLI.  I should probably update my project version, too. 🙂

**Comment by skylize** [comment]
Closing, as this is basically an upstream issue, which can be resolved through updating. Let us know if you have any more questions.

**Comment by Arcanemagus** [comment]
Actually, re-opening this since although the issue of the option not being available in their version of ESLint was an external error, the configuration not showing up was an error here.
